server/container_list.go

```diff
@@ -35,28 +37,29 @@ func (s *Server) filterContainerList(ctx context.Context, filter *pb.ContainerFi
 		if err != nil {
 			// If we don't find a container ID with a filter, it should not
 			// be considered an error.  Log a warning and return an empty struct
-			log.Warnf(ctx, "unable to find container ID %s", filter.Id)
-			return []*oci.Container{}
+			log.Warnf(ctx, "Unable to find container ID %s", filter.Id)
+			return nil
 		}
 		c := s.ContainerServer.GetContainer(id)
 		if c != nil {
 			switch {
 			case filter.PodSandboxId == "":
 				return []*oci.Container{c}
-			case c.Sandbox() == filter.PodSandboxId:
+			case strings.HasPrefix(c.Sandbox(), filter.PodSandboxId):
 				return []*oci.Container{c}
 			default:
-				return []*oci.Container{}
+				return nil
 			}
 		}
 	} else if filter.PodSandboxId != "" {
-		pod := s.ContainerServer.GetSandbox(filter.PodSandboxId)
-		if pod == nil {
-			return []*oci.Container{}
+		for _, sb := range s.ListSandboxes() {
```

Comment:
suggest `sandboxID, err := s.PodIDIndex().Get(filter.PodSandboxId)` to use the index vs scan

Reply:
Yes makes sense, thank you for the hint :)